tests/Events.elm

```diff
@@ -0,0 +1,84 @@
+module Events exposing (..)
+
+import Expect
+import Html exposing (Html, button, div, text, input)
+import Html.Attributes as Attr exposing (href)
+import Html.Events exposing (onClick, onInput)
+import Html.Lazy as Lazy
+import Test exposing (..)
+import Test.Html.Events as Events
+import Test.Html.Query as Query exposing (Single)
+import Test.Html.Selector exposing (tag)
+
+
+all : Test
+all =
+    describe "trigerring events"
+        [ test "returns msg for click on element" <|
+            \() ->
+                Query.fromHtml sampleHtml
+                    |> Query.findAll [ tag "button" ]
+                    |> Query.first
+                    |> Events.msgFor "click" "{}"
+                    |> Expect.equal (Ok SampleMsg)
+        , test "returns msg for click on lazy html" <|
+            \() ->
+                Query.fromHtml sampleLazyHtml
+                    |> Query.findAll [ tag "button" ]
+                    |> Query.first
+                    |> Events.msgFor "click" "{}"
+                    |> Expect.equal (Ok SampleMsg)
+        , test "returns msg for click on mapped html" <|
+            \() ->
+                Query.fromHtml sampleMappedHtml
+                    |> Query.findAll [ tag "button" ]
+                    |> Query.first
+                    |> Events.msgFor "click" "{}"
+                    -- TODO: Html.Map is ignored when traversing the DOM, need to fix it to avoid repeated mapping on tests like this
```

Comment:
🤔 I'm not super familiar with the underlying representation, but would `#2` from the original description address it? (With the understanding that that's a more invasive change.)